### PR TITLE
Smiles import minor adaptions

### PR DESCRIPTION
**Pull request by SamuelBehr**
Minor adaptions to the SMILES import: If no name for a structure is detected, the file name extended with the index of the line containing the SMILES code in the file (formerly extended with the count of so far parsed SMILES) is assigned as name to the structure; detected "invalid" lines are now being logged with their index in the file; the ParsableLinesCount and InvalidLinesCount are now only logged if "invalid" lines were encountered; added hyphen ("|") as possible SMILES file separator;

What do you two think of these changes?

Please review!

### CODE REVIEW

File: src/main/java/de/unijena/cheminf/mortar/model/io/Importer.java [hunk 1]
```diff
@@ -475,20 +478,28 @@ Checking for parsable SMILES code and saving the determined separator (if one is
...
                         tmpSmilesFileParsableLinesCounter++;
-                    } else {
-                        tmpSmilesFileInvalidLinesCounter++;
-                        continue;
                     }
-                } catch (InvalidSmilesException | IndexOutOfBoundsException anException) {  //case: invalid line or SMILES code
+                } catch (InvalidSmilesException | IndexOutOfBoundsException anException) {
+                    //case: invalid line or SMILES code
+                    //tmpContainsParsableSmilesCode stays false
+                }
```
[the remark this answers is not in the document]
Comment (SamuelBehr) [reply src/main/java/de/unijena/cheminf/mortar/model/io/Importer.java:498]: Hm, I see, I probably shouldn't do that. Would you prefer this here? :

                boolean tmpContainsParsableSmilesCode;         //removed the initialization
                try {
                    tmpProcessedLineArray = tmpSmilesFileNextLine.split(tmpSmilesFileDeterminedSeparator, 2);
                    if (!tmpProcessedLineArray[tmpSmilesCodeExpectedPosition].isEmpty()) {
                        tmpMolecule = tmpSmilesParser.parseSmiles(tmpProcessedLineArray[tmpSmilesCodeExpectedPosition]);
                        tmpContainsParsableSmilesCode = true;
                        tmpSmilesFileParsableLinesCounter++;
                    } else {
                        tmpContainsParsableSmilesCode = false;    //new
                    }
                } catch (InvalidSmilesException | IndexOutOfBoundsException anException) {
                    //case: invalid line or SMILES code
                    tmpContainsParsableSmilesCode = false;       //new
                }

[the remark this answers is not in the document]
Comment (JonasSchaub) [reply src/main/java/de/unijena/cheminf/mortar/model/io/Importer.java:498]: I guess the confusion comes from the expected SMILES position being empty (no exception) and the SMILES code being unparseable (exception) leading to the same code block which comes below. But yes, it is safer and more straightforward to understand to set these boolean values explicitly again in both cases, I'd say.

[the remark this answers is not in the document]
Comment (FelixBaensch) [reply src/main/java/de/unijena/cheminf/mortar/model/io/Importer.java:500]: That causes the confusion.
Anyway, I find it a bit unusual to place the logging outside the catch block. Why not like this and get rid of the Boolean:
```
try {
     tmpProcessedLineArray = tmpSmilesFileNextLine.split(tmpSmilesFileDeterminedSeparator, 2);
     String tmpSmiles = tmpProcessedLineArray[tmpSmilesCodeExpectedPosition].isBlank() ? null : 
                                     tmpProcessedLineArray[tmpSmilesCodeExpectedPosition];
     tmpMolecule = tmpSmilesParser.parseSmiles(tmpSmiles);
     tmpSmilesFileParsableLinesCounter++;
 }
catch (InvalidSmilesException | IndexOutOfBoundsException anException) {
     int tmpIndexInFile = tmpSmilesFileParsableLinesCounter + tmpSmilesFileInvalidLinesCounter;
     Importer.LOGGER.info("Contains no parsable SMILES string: line " + tmpIndexInFile + " (index).");
     tmpSmilesFileInvalidLinesCounter++;
     continue;
}
```

This should cause an InvalidSmilesException if position in the file is blank or null.

**Review by FelixBaensch — CHANGES_REQUESTED** [review]
See comments

**Review by JonasSchaub — APPROVED** [review]

**Comment by JonasSchaub** [comment]
@FelixBaensch I've made the changes you requested. Please review again (merging is blocked otherwise). Thank you!

**Review by FelixBaensch — APPROVED** [review]

**Pull request merged**